### ISSUE TITLE
Error when initializing client in Dapp - Unknown type, must be binary type

### ISSUE THREAD

**Issue by matthewcarlreetz** [WalletConnect/walletconnect-monorepo#1329]
**Describe the bug**
Initializing the `SignClient` on a React Native Dapp throws the error
> {"context": "client"} Unknown type, must be binary type

**SDK Version (if relevant)**
Client: React Native
Platform: `iOS 15.6`
Version:  
```json
{ 
    "@walletconnect/sign-client": "^2.0.0-rc.1",
    "react": "17.0.1",
    "react-native": "0.66.4"
}
```

**To Reproduce**
1. Initialize the client
    ```ts
    SignClient.init({
      projectId: Config.WALLET_CONNECT_PROJECT_ID,
      metadata: {
        name: 'Example Dapp',
        description: 'Example Dapp',
        url: '#',
        icons: ['https://walletconnect.com/walletconnect-logo.png'],
      },
    })

**Expected behavior**
Client initializes without error


**Smartphone (please complete the following information):**
 - Device: any iPhone
 - OS: tested with 15.6

**Additional context**
```bash
info Fetching system and libraries information...
System:
    OS: macOS 12.5
    CPU: (10) arm64 Apple M1 Max
    Memory: 2.07 GB / 64.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 17.8.0 - /opt/homebrew/bin/node
    Yarn: 1.22.19 - ~/Code/Wallet/nova-wallet-app/node_modules/.bin/yarn
    npm: 8.5.5 - /opt/homebrew/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK:
      API Levels: 28, 29, 30, 31
      Build Tools: 29.0.2, 30.0.2, 30.0.3, 31.0.0
      System Images: android-29 | Intel x86 Atom, android-29 | Intel x86 Atom_64, android-29 | Google APIs ARM 64 v8a, android-29 | Google APIs Intel x86 Atom, android-29 | Google APIs Intel x86 Atom_64, android-29 | Google Play ARM 64 v8a, android-29 | Google Play Intel x86 Atom, android-29 | Google Play Intel x86 Atom_64, android-31 | Google APIs ARM 64 v8a
      Android NDK: Not Found
  IDEs:
    Android Studio: 2021.2 AI-212.5712.43.2112.8609683
    Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_292 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1 
    react-native: 0.66.4 => 0.66.4 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

**Comment by xzilja** [comment]
@matthewcarlreetz will be looking at this today 👍 just need to set up env for react-native to test. Is hermes enabled here?

**Comment by matthewcarlreetz** [comment]
Were you able to reproduce @IljaDaderko ?

**Comment by matthewcarlreetz** [comment]
```
  use_react_native!(
    :path => config[:reactNativePath],
    :hermes_enabled => podfile_properties['expo.jsEngine'] == 'hermes'
  )

**Comment by matthewcarlreetz** [comment]
I haven't set `:hermes_enabled` to true

**Comment by finessevanes** [comment]
Is this still an issue?

**Comment by JeneaVranceanu** [comment]
Hello. Encountered a similar issue. 

<img width="802" alt="Screenshot 2023-02-21 at 11 39 21" src="https://user-images.githubusercontent.com/36865532/220308975-29cbefa8-d8a8-4679-b175-3a23a6122c9e.png">


```
Exception has occurred: Error: Unknown type, must be binary type 
{"type":"Buffer","data":[237,1,2,1,102,25,122,190,219,200,61,154,245,15,116,2,5,72,163,135,160,140,173,250,2,189,136,36,171,152,128,176,197,58]}
```

The issue seems to be in the `utils.js` of the `@walletconnect/relay-auth` (or rather `@walletconnect/ethereum-provider`). The `toString` function expects a `Uint8Array` but instead it gets a `Buffer`. 

<img width="1051" alt="Screenshot 2023-02-21 at 11 40 27" src="https://user-images.githubusercontent.com/36865532/220307569-77949d5f-919d-437b-a0ba-fb96aab4ed06.png">
<img width="603" alt="Screenshot 2023-02-21 at 11 41 00" src="https://user-images.githubusercontent.com/36865532/220307671-8c412614-e19c-497e-afa9-821a18e06851.png">

But since node v4.x `Buffer` are also the `Uint8Array` this error shouldn't arise - https://nodejs.org/docs/latest/api/buffer.html#buffers-and-typedarrays

-----

Update: so the issue is that `Buffer` will return `false` to `buffer instanceof Uint8Array` but it has all the APIs the `Uint8Array` does. In the `encode` function you see on the first screenshot even when we get the error if we try to execute manually in the console the `\`${ this.prefix }${ this.baseEncode(bytes) }\`` we get the expected output.

<img width="982" alt="Screenshot 2023-02-21 at 12 00 16" src="https://user-images.githubusercontent.com/36865532/220312819-585ddb69-107f-48d2-98eb-644519a120ec.png">

I think, this issue should be resolved on your side.

**Comment by JeneaVranceanu** [comment]
Found the issue - `uint8arrays` dependency fixed this issue in 3.1.1 but ethereum provider is using fixed version 3.1.0.

Here is the fix: https://github.com/achingbrain/uint8arrays/commit/017a456f7849700502d48b670c0caf5363854bcf#diff-40b2729f31c31777e2935b9b86b219f6de1dbd0ceb8ea1f69965a6337c75aabaR21

Here is the release with that fix: https://github.com/achingbrain/uint8arrays/releases/tag/v3.1.1

Here is the version ethereum provider is using: https://github.com/WalletConnect/walletconnect-monorepo/blob/v2.0/providers/ethereum-provider/package.json#L52
